src/main/java/org/mule/extension/oauth2/internal/AbstractGrantType.java

```diff
@@ -303,4 +305,15 @@ public boolean isDefaultTokenManager() {
     return defaultTokenManager || tokenManager == null;
   }
 
+  public boolean readsAuthenticatedResponseBody() {
+    java.util.Optional<String> refreshTokenWhenExpression = refreshTokenWhen.getLiteralValue();
```

Comment:
import `Optional`

Reply:
It collides with an annotation import org.mule.runtime.extension.api.annotation.param.Optional